src/cryptonote_core/cryptonote_core.cpp

```diff
@@ -1828,6 +1829,34 @@ bool core::check_tx_semantic(const transaction& tx, bool keeped_by_block) const
     return true;
 }
 //-----------------------------------------------------------------------------------------------
+void core::check_service_node_ip_address() {
+    if (!is_active_sn()) {
```

Comment:
I think this test would be worthwhile on any node running in service node mode, not only active.  I.e. if setting up a new service node you'd want to without having to wait until the registration actually goes through.